components/module-components/create-blogs/BlogListView.tsx

```diff
@@ -0,0 +1,106 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import Button from '../../reusable-components/button/Button';
+import ReusableTable from '../../reusable-components/reusable-table/ReusableTable';
+import Filter from '../SearchAndFilter/Filter';
+import Search from '../SearchAndFilter/Search';
+import ActionDropdown from '../../reusable-components/reusable-table/ActionDropdown';
+import style from './createBlogs.module.scss';
+
+interface BlogPost {
+  id: string;
+  title: string;
+  tags: { name: string }[];
+  author: string;
+}
+
+const BlogListView = () => {
+  const [data, setData] = useState<BlogPost[]>([]);
+  const url = process.env.NEXT_PUBLIC_API_BASE_URL + '/blog-posts'!;
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(url);
+        if (!response.ok) {
+          throw new Error(`${response.status} ${response.statusText}`);
+        }
+        const result = await response.json();
+        const transformedData: BlogPost[] = result.data.map(
+          (item: {
+            id: string;
+            title: string;
+            tags: { name: string }[];
+            author: { first_name: string; last_name: string };
+          }) => ({
+            id: item.id,
+            title: item.title,
+            tags: item.tags,
+            author: `${item.author.first_name} ${item.author.last_name}`,
+          })
```

Comment:
could you create interfaces for this data, so we can ensure type safety and improve code readability?